src/Development/IDE/Core/Shake.hs

```diff
@@ -164,6 +177,9 @@ data ShakeExtras = ShakeExtras
     ,exportsMap :: Var ExportsMap
     -- | A work queue for actions added via 'runInShakeSession'
     ,actionQueue :: ActionQueue
+    , hiedb :: HieDb -- ^ Use only to read
+    , hiedbWriter :: HieDbWriter -- ^ use to write
+    , persistentKeys :: Var (HMap.HashMap Key GetStalePersistent)
```

Comment:
doc comment.

Would it make sense to group these 3 new fields into a `HieDbState` record?